src/libraries/System.Console/src/System/Console.cs

```diff
@@ -392,42 +392,28 @@ public static int WindowTop
             set { ConsolePal.WindowTop = value; }
         }
 
-        [UnsupportedOSPlatform("android")]
-        [UnsupportedOSPlatform("browser")]
-        [UnsupportedOSPlatform("ios")]
-        [UnsupportedOSPlatform("tvos")]
         public static int WindowWidth
         {
+            [UnsupportedOSPlatform("android")]
+            [UnsupportedOSPlatform("browser")]
+            [UnsupportedOSPlatform("ios")]
+            [UnsupportedOSPlatform("tvos")]
             get { return ConsolePal.WindowWidth; }
+            [SupportedOSPlatform("windows")]
             set
-            {
-                if (Console.IsOutputRedirected)
-                {
-                    throw new IOException(SR.InvalidOperation_SetWindowSize);
-                }
-
-                ArgumentOutOfRangeException.ThrowIfNegativeOrZero(value, nameof(WindowWidth));
-
-                ConsolePal.WindowWidth = value;
```

Comment:
We don't still want the same up front validation across all OSes (even if after this it'll throw PNSE? Same below.

Reply:
I don't have an opinion.

I moved the validation into the Windows PAL because that is where it was before Unix support got added.